### PR TITLE
Feat(eos_cli_config_gen): Add aaa authorization policy and dynamic

### PR DESCRIPTION
**Pull request by tgodaA**
## Change Summary

Add the aaa authorization `policy` and `dynamic` commands.

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

```yaml
aaa_authorization:
  policy:
    local_default_role: < role_name >
  dynamic:
    dot1x_additional_groups:
      - < group_name >
```

## How to test

`pre-commit run -a`
`molecule converge -s eos_cli_config_gen -- --limit aaa`

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by tgodaA** [comment]
Tested commands on a device:

image

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-authorization.j2 [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
 {% if aaa_authorization is arista.avd.defined %}
+{%     if aaa_authorization.policy.local_default_role is arista.avd.defined %}
+aaa authorization policy local default-role {{ aaa_authorization.policy.local_default_role }}
+{%     endif %}
 {%     if aaa_authorization.serial_console is arista.avd.defined(true) %}
 aaa authorization serial-console
 {%     endif %}
+{%     if aaa_authorization.dynamic.dot1x_additional_groups is arista.avd.defined %}
+aaa authorization dynamic dot1x additional-groups group {{ aaa_authorization.dynamic.dot1x_additional_groups | join(" group ") }}
+{%     endif %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-authorization.j2:11]: @ClausHolbechArista is there a way to protect this logic from empty list in schemas? Else I need to add a test to check null groups ( length > 0)

Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-authorization.j2:11]: @tgodaA you can add `min_length: 1` in the schema to trigger validation error:
https://avd.sh/en/stable/docs/input-variable-validation-BETA.html#schema-options-for-type-list-list

Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/aaa-authorization.j2:11]: Great, thank you.

**Review by emilarista — APPROVED** [review]
LGTM!

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**